Client/mods/deathmatch/logic/luadefs/CLuaPlayerDefs.cpp

```diff
@@ -644,3 +648,168 @@ bool CLuaPlayerDefs::IsPlayerCrosshairVisible()
 {
     return g_pGame->GetHud()->IsCrosshairVisible();
 }
+
+bool CLuaPlayerDefs::SetPlayerHudComponentProperty(eHudComponent component, eHudComponentProperty property, std::variant<CVector2D, float, bool> value)
+{
+    switch (property)
+    {
+        case eHudComponentProperty::POSITION:
+        {
+            if (!std::holds_alternative<CVector2D>(value))
+                return false;
+
+            g_pGame->GetHud()->SetComponentPosition(component, std::get<CVector2D>(value));
+            return true;
+        }
+        case eHudComponentProperty::SIZE:
+        {
+            if (!std::holds_alternative<CVector2D>(value))
+                return false;
+
+            g_pGame->GetHud()->SetComponentSize(component, std::get<CVector2D>(value));
+            return true;
+        }
+        case eHudComponentProperty::FILL_COLOR:
+        {
+            switch (component)
+            {
+                case HUD_HEALTH:
+                case HUD_ARMOUR:
+                case HUD_BREATH:
+                {
+                    if (!std::holds_alternative<float>(value))
+                        return false;
+
+                    g_pGame->GetHud()->SetComponentBarColor(component, std::get<float>(value));
```

Comment:
Why do we need to store a color as floaing point number?

Reply:
Any number passed from Lua through the parser is actually always of type float (Lua doesn't have types like int to represent numbers). Therefore, reading the number as a float is not a problem, although it can also be read as an int if needed.

Reply:
No matter how numbers are represented in Lua, let's just keep it on a low level. We have a modern Lua parser that gives us integer types which must be used. Floating point types are unstable and they are never used for representing exact values. uint32_t is recognized standard for RGBA8 colors and must be used for this purpose.

Reply:
For the parser, it must be int32_t because, for some reason, tocolor returns a negative value

Reply:
Since `BLINKING_HP_VALUE` also requires **float** and to prevent a Lua parser from being confused (as **float** and **uint32** have similar semantics) I recommend you to change `SetComponentBarColor` signature leaving `std::variant<CVector2D, float, bool>` as is.